backend/src/apiserver/resource/resource_manager.go

```diff
@@ -1967,7 +1967,17 @@ func (r *ResourceManager) GetArtifactSessionInfo(ctx context.Context, artifact *
 	}
 
 	// Retrieve Session info
-	sessionInfoString := artifactCtx.CustomProperties["store_session_info"].GetStringValue()
+	storeSessionInfo, ok_session := artifactCtx.CustomProperties["store_session_info"]
+	bucketSessionInfo, ok_bucket := artifactCtx.CustomProperties["bucket_session_info"]
+
+	var sessionInfoString = ""
+
+	if ok_session {
+		sessionInfoString = storeSessionInfo.GetStringValue()
+	} else if ok_bucket {
+		sessionInfoString = bucketSessionInfo.GetStringValue()
+	}
```

Comment:
I would like to see the logic work such that, we only evaluate for 

`artifactCtx.CustomProperties["bucket_session_info"]` if `!ok_session` 

also, this is currently not enough because the format of `bucket_session_info` is also different than `store_session_info` so you need to parse this into a sessionInfo manualy

try running an example pipeline with iris, and inspect the contextProperty to see the value of `bucket_session_info`